test/Microsoft.Extensions.Primitives.Tests/StringValuesTests.cs

```diff
@@ -302,5 +333,97 @@ public void Concat(StringValues stringValues, string[] array)
             string[] expectedAppended = filled.Concat(array).ToArray();
             Assert.Equal(expectedAppended, StringValues.Concat(new StringValues(filled), stringValues));
         }
+
+        [Fact]
+        public void Equals_Operators()
+        {
+            var equalString = "abc";
+            var equalStringArray = new string[] { equalString };
+            var equalStringValues = new StringValues(equalString);
+
+            var notEqualString = "bcd";
+            var notEqualStringArray = new string[] { notEqualString };
+            var notEqualStringValues = new StringValues(notEqualString);
+
+            var stringArray = new string[] { "abc", "bcd" };
+            var stringValuesArray = new StringValues(stringArray);
+
+            Assert.True(equalStringValues == equalStringValues);
+            Assert.True(equalStringValues == equalString);
+            Assert.True(equalString == equalStringValues);
+            Assert.True(equalStringValues == equalStringArray);
+            Assert.True(equalStringArray == equalStringValues);
+
+            Assert.False(equalStringValues != equalStringValues);
+            Assert.True(equalStringValues != notEqualStringValues);
+            Assert.True(equalStringValues != notEqualString);
+            Assert.True(notEqualString != equalStringValues);
+            Assert.True(equalStringValues != notEqualStringArray);
+            Assert.True(notEqualStringArray != equalStringValues);
```

Comment:
This is looking better overall. The inputs to this and `Equals_Instance` and `Equals_Object` should be using `[Theory]`

`==` and `!=` should be separate tests using the same inputs but the opposite assert